src/lib/tabs/tab-nav-bar/tab-nav-bar.ts

```diff
@@ -50,8 +59,14 @@ export class MdTabLink {
 @Directive({
   selector: '[md-tab-link]',
 })
-export class MdTabLinkRipple extends MdRipple {
+export class MdTabLinkRipple extends MdRipple implements OnDestroy {
   constructor(private _element: ElementRef) {
     super(_element);
   }
+
+  // In certain cases the parent destroy handler
+  // may not get called. See Angular issue #11606.
+  ngOnDestroy() {
+    super.ngOnDestroy();
+  }
```

Comment:
I couldn't find a good way of doing it and opted not to since the ripple didn't have a test for it either. Should I spend some time adding it to both places?